orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -65,10 +65,28 @@ public String name() {
     public TypeDescription type() {
       return type;
     }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      OrcField orcField = (OrcField) o;
+      return Objects.equals(name, orcField.name) && Objects.equals(type, orcField.type);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(name, type);
```

Comment:
Why was it necessary to add `equals` and `hashCode` implementations to `ORCField`? I don't see it used in a map or comparison anymore. Maybe this can be removed?